tests/integration/pkg/hooks/ext-auth.go

```diff
@@ -0,0 +1,110 @@
+package hooks
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/avast/retry-go/v4"
+	k8sclient "github.com/kyma-project/api-gateway/tests/integration/pkg/client"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/manifestprocessor"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/resource"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/testcontext"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+	"time"
+)
+
+//go:embed manifests/istio-cr.yaml
+var istioCrManifest []byte
+
+//go:embed manifests/ext-auth.yaml
+var extAuthManifests []byte
+
+func applyExtAuthorizerIstioCR() error {
+	log.Printf("Apply Istio CR with External Authorizer config")
+	istioCr, err := getIstioCr()
+	if err != nil {
+		return err
+	}
+	client := k8sclient.GetK8sClient()
+	spec := istioCr.Object["spec"]
+
+	return retry.Do(func() error {
+		_, err := controllerutil.CreateOrPatch(context.Background(), client, &istioCr, func() error {
+			istioCr.Object["spec"] = spec
+			return nil
+		})
+		return err
+	}, []retry.Option{
+		retry.Delay(time.Duration(2) * time.Second),
+		retry.Attempts(5),
+		retry.DelayType(retry.FixedDelay),
+	}...)
+}
+
+func getIstioCr() (unstructured.Unstructured, error) {
```

Comment:
I'd move this function to `istio.go`, because this is where I'd look for a function like that.